### ISSUE TITLE
Unable to open url from terminal

### ISSUE THREAD

**Issue by mobid** [atlas-engineer/nyxt#2804]
**Describe the bug**

When I have running instance of nyxt, nothing happens, when I try to open url (or eval expression) from terminal.

**Precise recipe to reproduce the issue**

```
me@host ~/s/nyxt (master)> ./nyxt -I & # New instance without config
me@host ~/s/nyxt (master)> ./nyxt https://ddg.gg # Try to open url
Nyxt version 3-pre-release-2-963-g842fe56d
<INFO> [14:21:56] Source location: #P"/home/me/src/nyxt/"
# Nothing happens
```

It works, if I write directly into socket:
```
me@host ~/s/nyxt (master)> echo -n '(nil "https://ddg.gg")' | nc -U /run/user/1000/nyxt/nyxt.socket
^C⏎ 
<INFO> [14:29:36] Externally requested URL(s): https://ddg.gg
```

**Information**

- OS name+version: NixOS unstable
- Graphics card and driver: nvidia
- Desktop environment / Window manager name+version: none+xmonad
- How you installed Nyxt (Guix pack, package manager, build from source): nix-shell build-scripts/shell.nix --run 'make nyxt'
- Information from `show-system-information`:

```
Nyxt version: 3-pre-release-2-963-g842fe56d
Renderer: GI-GTK
Operating system kernel: Linux 5.15.91
Lisp implementation: SBCL 2.3.0.nixos (Dynamic space size: 3221225472)
Features: (:NYXT-GI-GTK :NYXT-GTK :NYXT-GI-GTK :NYXT-GTK :NYXT-3.1 :NYXT-3
 :NYXT-3-PRE-RELEASE-2 :NYXT-3-PRE-RELEASE-2-963-G842FE56D :SLYNK :PLUMP-UTF-32
 :PARENSCRIPT :NSYMBOLS :FSET-EXT-STRINGS :NJSON :GLOBAL-VARS :DECLARE-TYPES
 :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES :SWANK :CL-FAD :LPARALLEL
 :21BIT-CHARS :CUSTOM-HASH-TABLE-NATIVE :CL-PPCRE-UNICODE :CL-UNICODE
 :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED :CL-PPCRE
 :CHUNGA :FLEXI-STREAMS :WEBKIT2 :WEBKIT2-2.38 :WEBKIT2-CORS-ALLOWLIST
 :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING :WEBKIT2-MUTE :WEBKIT2-EMOJI
 :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22 :GTK-3-20 :GTK-3-18 :GTK-3-16
 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6 :GTK-3-4 :GTK :GDK-3-22
 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12 :GDK-3-10 :GDK-3-8 :GDK-3-6
 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF :CLOSER-MOP :GLIB-2-30 :GLIB-2-32
 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42 :GLIB-2-44 :GLIB-2-46
 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56 :GLIB-2-58 :GLIB
 :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS
 :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE
 CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64 CFFI-FEATURES:UNIX :CFFI
 CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP :FAST-IO-SV :FAST-IO
 :CL-JSON-CLOS :CL-JSON :SBCL-USES-SB-ROTATE-BYTE CHIPZ-SYSTEM:GRAY-STREAMS
 :THREAD-SUPPORT :ASDF3.3 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX
 :NON-BASE-CHARS-EXIST-P :ASDF-UNICODE :ARENA-ALLOCATOR :X86-64 :GENCGC :64-BIT
 :ANSI-CL :COMMON-LISP :ELF :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN
 :PACKAGE-LOCAL-NICKNAMES :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS
 :SB-THREAD :SB-UNICODE :SBCL :UNIX)

ASDF version: 3.3.1
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY)
Critical dependencies: (/home/me/src/nyxt/_build/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /home/me/src/nyxt/_build/cl-gobject-introspection/cl-gobject-introspection.asd
 /home/me/src/nyxt/_build/cl-webkit/webkit2/cl-webkit2.asd)
```

**Output when started from a shell**

```
me@host ~/.l/s/nyxt> ./nyxt -I
Nyxt version 3-pre-release-2-963-g842fe56d                                                                                                         [ nyxt-dev ]
<INFO> [14:21:16] Source location: #P"/home/me/src/nyxt/"
<INFO> [14:21:16] Listening to socket: #P"/run/user/1000/nyxt/nyxt.socket"
<INFO> [14:21:16] GTK extensions directory: #P"/home/me/src/nyxt/libraries/web-extensions/"
<INFO> [14:21:16] GTK extensions directory: #P"/home/me/src/nyxt/libraries/web-extensions/"
Could not determine the accessibility bus address
<INFO> [14:21:17] Loading #P"/home/me/.local/share/nyxt/history/default.lisp".
<INFO> [14:21:17] Loading #P"/home/me/.local/share/nyxt/auto-rules.lisp".
```

**Comment by aartaka** [comment]
@mobid, to talk (`--eval` and `--load`) to remote instance you need to use `--remote` CLI arg and `remote-execution-p` set to T in the remote instance:
``` lisp
(define-configuration browser
  ;; Enable --remote --eval code evaluation.
  ((remote-execution-p t)))
```

But we didn't anticipate the case of the remote URL-loading, it seems. @Ambrevar, it tried to hack it up, but it's not a five-minute hack it seems.

In the meanwhile @mobid you can use
``` sh
nyxt --remote --eval '(make-buffer :url "https://ddg.gg")' --quit
# or, to load it in the current buffer:
nyxt --remote --eval '(buffer-load "https://ddg.gg")' --quit
```

Also see `nyxt:manual#scripting` manual entry if you're on 3.*

**Comment by aartaka** [comment]
@mobid, so, it seems that the thing you're asking for actually was there. Closing then?

If not, what's your use-case?

**Comment by mobid** [comment]
Problem is, that third programs cannot open urls in nyxt.

I have tried again: (with nyxt 3 build from source)
```sh
# Run main instance
$ nix-shell build-scripts/shell.nix --run './nyxt -I'
```

```sh
$ nix-shell build-scripts/shell.nix
$ ./nyxt -v -I https://ddg.gg
Arguments parsed: (VERBOSE T NO-CONFIG T URLS (https://ddg.gg)) and (https://ddg.gg)
Nyxt version 3-pre-release-2-963-g842fe56d
 <INFO> [10:20:50] nyxt start.lisp (start-browser) -
  Source location: #P"/home/me/src/nyxt/"
# ... nothing happens
# But this works:
$ ./nyxt -v -I --remote -e '(nil "https://ddg.gg")' --quit
Arguments parsed: (VERBOSE T NO-CONFIG T REMOTE T EVAL (nil "https://ddg.gg")
                   QUIT T URLS NIL) and NIL
 <INFO> [10:20:53] nyxt start.lisp (load-or-eval) -
  Probing remote instance listening to /run/user/1000/nyxt/nyxt.socket.
# ... main instance open page
```

I have browsing around the code and it seems, that execution stops with this condition: https://github.com/atlas-engineer/nyxt/blob/master/source/start.lisp#L583

**Comment by aartaka** [comment]
@Ambrevar, I can reproduce this. Seems like some codepath is behaving wrong when:
- There's an instance with `remote-execution-p` set to T.
- Socket exists.
- CLI call of `nyxt URL` (without `--remote`) happens.

**Comment by Ambrevar** [comment]
I didn't quite understand the recipe.  What I could reproduce is that `nyxt URL` would do nothing if another Nyxt instance was up.

#2810 should fix this.

**Comment by Ambrevar** [comment]
Fixed with 6457714f6e79857307d12d7962e21c8d1abf04c7.
